hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieClusteringConfig.java

```diff
@@ -143,6 +143,15 @@ public class HoodieClusteringConfig extends HoodieConfig {
       .sinceVersion("0.9.0")
       .withDocumentation("Config to control frequency of async clustering");
 
+  public static final ConfigProperty<Integer> CLUSTERING_MAX_THREADS = ConfigProperty
+      .key("hoodie.clustering.max.threads")
+      .defaultValue(10)
+      .sinceVersion("0.13.0")
+      .withDocumentation("Maximum number of parallelism jobs submitted in clustering operation. "
+          + "If the resource is sufficient(Like Spark engine has enough idle executors), increasing this "
+          + "value will let the clustering job run faster, while it will give additional pressure to the "
+          + "execution engines to manage more concurrent running jobs.");
+
```

Comment:
In my opinion, the increase of parameters will increase the difficulty of users; and I  think 10 may be too small, the value can be determined by the number of CPU cores or the spark parallelism, It may be better, and the number of cluster groups can be used as the upper limit.

Reply:
Nice suggestion, can we change it as above-mentioned style?

Reply:
It's difficult to get `the value can be determined by the number of CPU cores or the spark parallelism`, and I think users could set values larger than CPU cores, we need to support it.

Added the number of cluster groups can be used as the upper limit.

Reply:
Where does the 15 come from?

Reply:
Delta has a configure to control the concurrency when optimizing file layout, `optimize.maxThreads`, and its default value is 15

Reply:
Fine with the change, do you think we should use `hoodie.clustering.max.parallelism` instead? We have `hoodie.upsert.shuffle.parallelism` in write config, it seems `parallelism` is the terminology for parallel execution in Hudi.

Reply:
make sense, let me change it